File: helpdesk/src/Helpdesk.Models/DocumentModel.cs
namespace Helpdesk.Models
{
    public class DocumentModel
    {
        public int? Id { get; set; }

        public int? BlobId { get; set; }

        public string NoteFileName { get; set; }

        public string NoteFileType { get; set; }

        public byte[] NoteContents { get; set; }

        public string Description { get; set; }

        public bool? Deleted { get; set; }
    }

    public class DocumentViewModel : DocumentModel, IBlobDownloadable
    {
        public long UnixTimeSeconds { get; set; }

        /// <summary>
        /// // Url-safe Base64 / RFC 4648
        /// https://tools.ietf.org/html/rfc4648
        /// </summary>
        public string Hmac { get; set; }

        public string BlobServiceUrl { get; set; }
    }
}

File: helpdesk/src/Helpdesk.SPA/src/models/documentModel.js
export class DocumentModel {
  constructor(obj = {}) {
    this.id = obj.id || null;
    this.noteFileName = obj.noteFileName || '';
    this.noteFileType = obj.noteFileType || '';
    this.section = obj.section || '';
    this.description = obj.description || '';
    this.blobId = obj.blobId || null;
    this.unixTimeSeconds = obj.unixTimeSeconds || 0;
    this.hmac = obj.hmac || '';
    this.blobServiceUrl = obj.blobServiceUrl || '';
    this.deleted = obj.deleted || false;
    this.uid = obj.uid || obj.id;
  }
}
